ome/ome-zarr-py
# Add --downsample_z option to ome_zarr scale

This adds support for Z-downsampling to the `ome_zarr scale` command.
To avoid loading large amounts of 5D data into memory, the Z scaling happens independently of XY downsampling.
XY downsampling is performed on one XY 2D-plane at a time (with the default 'nearest' scaling), writing each one to disk in turn. The Z downsampling is then performed on these smaller volumes, scaling a 3D volume and writing this to disk for each T and C index in turn.

To test:

 - Scale a zarr array to a pyramid, downsampling in X and Y (no change ).

```
$ ome_zarr scale 3851.zarr/0 3851_pyramid.zarr
```

 - Scale an existing pyramid to a new pyramid:
     -  Each array except '0' downsampled in Z.
     - Chunks are no-longer single Z/C/T
     - '0' array copied over.
     - Top level `.zattrs` copied over if present.
     - This allows the output of `$ omero zarr export Image:1` to be downsampled in Z

```
$ ome_zarr scale 3851.zarr 3851_zpyramid.zarr --downsample_z
```

  - Scale an array to pyramid (down-sample X, Y and Z)
     - this combines the 2 operations above
     - writes data to a _temp pyramid directory to pass data from 1st to 2nd - (_temp then deleted)

```
$ ome_zarr scale 3851.zarr/0 3851_pyramid.zarr --downsample_z
```

TODO?:
 - Pyramid `arrays` generated above are in *new* pyramid `groups`.  We could add options for adding them alongside existing `arrays` in the same group - and updating the group `.zattrs` `multiscales` info?
 - But we might first want to decide on any spec changes for `multiscales` e.g. see https://forum.image.sc/t/ome-ngff-combine-multiple-images-into-one-dataset/47619/14
 - Do we want to be able to choose chunk sizes for scaled arrays? Currently size Z, C, T = 1
